src/multicall.rs

```diff
@@ -0,0 +1,133 @@
+use alloy::network::{ReceiptResponse, TransactionBuilder};
+use alloy::providers::Provider;
+use alloy::rpc::types::TransactionRequest;
+use alloy::{
+    providers::{Network, RootProvider},
+    transports::{Transport, TransportErrorKind, TransportResult},
+};
+use alloy_primitives::{Address, Bytes};
+use alloy_sol_types::{sol, SolCall};
+use futures::future::join_all;
+use oneshot::{Receiver, Sender};
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, Instant};
+use tokio::time::sleep;
+
+// Define the Multicall3 interface
+sol! {
+    interface Multicall3 {
```

Comment:
```suggestion
    #[sol(rpc)]
    interface Multicall3 {
```
https://github.com/alloy-rs/examples/blob/main/examples/contracts/examples/interact_with_abi.rs